packages/dd-trace/test/appsec/iast/taint-tracking/taint-tracking-impl.spec.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+
+const { testThatRequestHasVulnerability, copyFileToTmp } = require('../utils')
+const { storage } = require('../../../../../datadog-core')
+const iastContextFunctions = require('../../../../src/appsec/iast/iast-context')
+const { newTaintedString, isTainted } = require('../../../../src/appsec/iast/taint-tracking/operations')
+const { clearCache } = require('../../../../src/appsec/iast/vulnerability-reporter')
+const { expect } = require('chai')
+
+const propagationFns = [
+  'concatSuffix',
+  'insertStr',
+  'appendStr',
+  'trimStr',
+  'trimStartStr',
+  'trimEndStr'
+]
+
+const commands = [
+  '  ls -la  ',
+  '  ls -la',
+  'ls -la  ',
+  'ls -la',
+  ' ls -la  人 ',
+  ' ls -la  𠆢𠆢𠆢 ',
+  ' ls -ls �'
+]
+
+const propagationFunctionsFile = path.join(__dirname, 'resources/propagationFunctions.js')
+
+describe('TaintTracking', () => {
+  let instrumentedFunctionsFile
+  beforeEach(() => {
+    instrumentedFunctionsFile = copyFileToTmp(propagationFunctionsFile)
+  })
+
+  afterEach(() => {
+    fs.unlinkSync(instrumentedFunctionsFile)
+    clearCache()
+  })
+
+  describe('should propagate strings', () => {
+    propagationFns.forEach((propFn) => {
+      describe(`using ${propFn}()`, () => {
+        commands.forEach((command) => {
+          describe(`with command: '${command}'`, () => {
+            testThatRequestHasVulnerability(function () {
+              const store = storage.getStore()
+              const iastContext = iastContextFunctions.getIastContext(store)
+              const commandTainted = newTaintedString(iastContext, command, 'param', 'Request')
+
+              const propFnInstrumented = require(instrumentedFunctionsFile)[propFn]
+              const propFnOriginal = require(propagationFunctionsFile)[propFn]
```

Comment:
Can we do this require in line 33? At this way we will require it only once and out of the request/iast context.